docs/advanced/mbed_targets.md

```diff
@@ -161,13 +165,30 @@ class TEENSY3_1Code:
             binh.tofile(f, format='hex')
 ```
 
-In this case, it converts the output file (`binf`) from binary format to Intel HEX format.
+The `post_build_hook` for the `TEENSY3_1` converts the output file (`binf`) from binary format to Intel HEX format. See the other hooks in `tools/targets/__init__.py` for more examples of hook code. 
 
-The hook code can look quite different for different targets. Take a look at the other classes in *targets.py* for more examples of hook code.
+## `device_name`
 
-## device_name
-
-This property is used to pass necessary data for exporting the mbed code to various third party tools and IDEs.
+Use this property to pass necessary data for exporting to various third party tools and IDEs and for building applications with bootloaders.
 
 Please see [exporters.md](exporters.md) for information about this field.
 
+## `OUTPUT_EXT`
+
+The `OUTPUT_EXT` property controls the file type emitted for a target by the build system. You may set `OUTPUT_EXT` to `bin` for binary format, `hex` for Intel HEX format and `elf` for ELF format. We discourage using the `elf` value for `OUTPUT_EXT` because the build system must always emit an ELF file. 
+
+## `default_lib`
+
+The `delault_lib` property controls which library, small or standard, the `GCC_ARM` toolchain links. The `default_lib` property may take on the values `std` for the standard library and `small` for the reduced size library.
+
+## `bootloader_supported`
+
+The `bootloader_supported` property controls whether the build system allows a bootloader or a bootloader-using application to be built for a target. The default value of `bootloader_supported` is `false`.
+
+## `release_versions`
+
+The `release_versions` property is a list of major versions of mbed OS that the target supports. The list within `release_versions` may only contain `2`, indicating that the support of mbed OS 2, and `5`, indicating the support of mbed OS 5. Any target a `release_version` with a `2` in its list will be built as a static library as part of the mbed OS 2 release process.
```

Comment:
Could you please clarify the last sentence? I'm having trouble understanding it.